code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm

```diff
@@ -170,8 +188,90 @@
 	var/health_percent = round((max_integrity / obj_integrity) * 100)
 	switch(health_percent)
 		if(-INFINITY to 25)
-			to_chat(parent, "<span class='xenohighdanger'>Your core is under attack, and dangerous low on health!</span>")
+			to_chat(parent, "<span class='xenohighdanger'>Your core is under attack, and is dangerously low on health!</span>")
 		if(26 to 75)
-			to_chat(parent, "<span class='xenodanger'>Your core is under attack, and low on health!</span>")
+			to_chat(parent, "<span class='xenodanger'>Your core is under attack, and is low on health!</span>")
 		if(76 to INFINITY)
 			to_chat(parent, "<span class='xenodanger'>Your core is under attack!</span>")
+
+/mob/living/carbon/xenomorph/hivemind/upgrade_xeno(newlevel) //where we also upgrade the max health of our core and restore it to maximum health
+	. = ..()
+	to_chat(src, "<span class='xenonotice'>Our core strengthens, and becomes more resilient!</span>")
+	core.max_integrity = src.xeno_caste.core_maximum_hitpoints
+	core.obj_integrity = core.max_integrity
+
+/mob/living/carbon/xenomorph/proc/hivemind_weed_vibecheck()
+	if(!src) //Sanity
```

Comment:
make this either a hivemind proc or a ability proc and rename it to something more comprehensible

Reply:
Done.